### PR TITLE
Migrate project to use Jest

### PR DESCRIPTION
**Pull request by frederikprijck**
### Description

Updates the project to use Jest, aligning it with our other SDKs such as Auth0-SPA-JS and Auth0-React.

### Testing

- [x] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [x] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not the default branch

### CODE REVIEW

File: projects/auth0-angular/src/lib/auth.interceptor.spec.ts
Comment (frederikprijck) [inline projects/auth0-angular/src/lib/auth.interceptor.spec.ts:60]: It looks like Jest handles these Http tests a bit differently, this made it work for all tests, but had to make the function to be async and await it in every test.

Before merging this I will have another look and see if we can avoid this. But for now, this seems to be what solves these tests to run successfully with Jest.

File: projects/playground/src/app/app.component.spec.ts
Comment (frederikprijck) [inline projects/playground/src/app/app.component.spec.ts:122]: Jest uses a different port than Karma did.

File: projects/playground/src/app/app.component.spec.ts
Comment (frederikprijck) [inline projects/playground/src/app/app.component.spec.ts:215]: We can't cast to jest.SpyInstance without casting to unknown first. This could be approved upon in a future PR if needed.

File: projects/auth0-angular/src/lib/auth.service.spec.ts
Comment (frederikprijck) [inline projects/auth0-angular/src/lib/auth.service.spec.ts:100]: Our service does a lot of things in its constructor, meaning it makes sense to have the tests control the creation. Some tests might be fine to have the service created outside of it, but previously we were creating the service outside of every test, and recreating it inside some specific tests.

Important to know is that services are Singletons, so creating the service in the tests doesnt create a new instance if one was created outside of it.

Therefore, all tests now create the instance of the service themselves.

File: projects/auth0-angular/src/lib/auth.service.spec.ts
Comment (frederikprijck) [inline projects/auth0-angular/src/lib/auth.service.spec.ts:627]: It looks like this test needs to be adapted to work with Jest's JSDom, we set the search value on the window's location.

**Review by ewanharris — APPROVED** [review]

**Pull request merged**
